plugins/ingest/src/main/java/org/elasticsearch/ingest/Pipeline.java

```diff
@@ -70,6 +70,21 @@ public String getDescription() {
         return processors;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Pipeline pipeline = (Pipeline) o;
+        return Objects.equals(id, pipeline.id) &&
+                Objects.equals(description, pipeline.description) &&
+                Objects.equals(processors, pipeline.processors);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, description, processors);
```

Comment:
Is it worth to have specific unit tests for each equals and hashcode that we are adding here? Just make sure that the two are consistent etc.